habitat_sim/agent/controls/default_controls.py

```diff
@@ -18,67 +25,79 @@
 _Z_AXIS = 2
 
 _rotate_local_fns = [
-    hsim.SceneNode.rotate_x_local,
-    hsim.SceneNode.rotate_y_local,
-    hsim.SceneNode.rotate_z_local,
+    SceneNode.rotate_x_local,
+    SceneNode.rotate_y_local,
+    SceneNode.rotate_z_local,
 ]
 
 
-def _move_along(scene_node: hsim.SceneNode, distance: float, axis: int):
+def _move_along(scene_node: SceneNode, distance: float, axis: int):
     ax = scene_node.transformation[axis].xyz
     scene_node.translate_local(ax * distance)
 
 
-def _rotate_local(scene_node: hsim.SceneNode, theta: float, axis: int):
+def _rotate_local(scene_node: SceneNode, theta: float, axis: int):
     _rotate_local_fns[axis](scene_node, mn.Deg(theta))
     scene_node.rotation = scene_node.rotation.normalized()
 
 
+def _apply_look_constraint(scene_node: SceneNode, constraint: float):
+    constraint = mn.Deg(constraint)
+    rotation = scene_node.rotation
+    look_axis = rotation.transform_vector(mn.Vector3(0, 0, -1))
+    look_angle = mn.Rad(np.arctan2(look_axis[1], -look_axis[2]))
+
+    if look_angle > constraint:
```

Comment:
Might be simpler if _apply_look_constraint took constraint and actuation_spec.amount as input and returned the final rotation amount use for _rotate_local resulting in only one call to _rotate_local vs two.